src/libopensc/card-cardos.c

```diff
@@ -230,7 +237,7 @@ static int cardos_init(sc_card_t *card)
 		_sc_card_add_rsa_alg(card, 2048, flags, 0);
 	}
 
-	if (card->type == SC_CARD_TYPE_CARDOS_V5_0) {
+	if (card->type >= SC_CARD_TYPE_CARDOS_V5_0) {
```

Comment:
Please use an explicit test (e.g. in a switch statement) instead of assuming a specific order of the card types.